core/domain/wipeout_service.py

```diff
@@ -315,103 +410,303 @@ def _delete_models(user_id, user_role, module_name):
+    if snapshot_metadata_ids != commit_log_ids:
         logging.error(
             'The commit log and snapshot %s IDs differ. '
             'Snapshots without commit logs: %s, '
             'Commit logs without snapshots: %s.',
-            activity_category,
-            list(activity_ids - commit_log_ids),
-            list(commit_log_ids - activity_ids))
-
-    activity_ids |= commit_log_ids
-    # The activity_mappings field might have only been partially generated, so
-    # we fill in the missing part for this activity category.
-    if activity_category not in pending_deletion_request.activity_mappings:
-        pending_deletion_request.activity_mappings[activity_category] = (
-            _generate_activity_to_pseudonymized_ids_mapping(activity_ids))
-        save_pending_deletion_requests([pending_deletion_request])
+            entity_category,
+            list(snapshot_metadata_ids - commit_log_ids),
+            list(commit_log_ids - snapshot_metadata_ids))
+
+    _save_entity_mappings(
+        pending_deletion_request,
+        entity_category,
+        snapshot_metadata_ids | commit_log_ids)
+
+    return (snapshot_metadata_models, commit_log_models)
+
+
+def _pseudonymize_entity_models(
+        pending_deletion_request,
+        entity_category,
+        snapshot_model_class,
+        commit_log_model_class,
+        commit_log_model_field_name):
+    """Pseudonymize the entity models for the user with user_id.
+
+    Args:
+        pending_deletion_request: PendingDeletionRequest. The pending deletion
+            request object for which to pseudonymize the models.
+        entity_category: models.NAMES. The category of the models that are
+            being pseudonymized.
+        snapshot_model_class: class. The metadata model class that is being
+            pseudonymized.
+        commit_log_model_class: class. The commit log model class that is being
+            pseudonymized.
+        commit_log_model_field_name: str. The name of the field holding the
+            entity id in the corresponding commit log model.
+    """
+    snapshot_metadata_models, commit_log_models = (
+        _collect_entity_ids_from_snapshots_and_commit(
+            pending_deletion_request,
+            entity_category,
+            [snapshot_model_class],
+            commit_log_model_class,
+            commit_log_model_field_name
+        )
+    )
 
-    def _pseudonymize_models(activity_related_models, pseudonymized_user_id):
+    def _pseudonymize_models(entity_related_models, pseudonymized_id):
         """Pseudonymize user ID fields in the models.
 
         This function is run in a transaction, with the maximum number of
-        activity_related_models being MAX_NUMBER_OF_OPS_IN_TRANSACTION.
+        entity_related_models being MAX_NUMBER_OF_OPS_IN_TRANSACTION.
 
         Args:
-            activity_related_models: list(BaseModel). Models whose user IDs
+            entity_related_models: list(BaseModel). Models whose user IDs
```

Comment:
Ah ... sorry, @vojtechjelinek, but I think "entity" is actually worse from a naming perspective than "activity". We use it in a really general sense already (e.g. EntityIdNotFoundError) so this makes it sound like it's relating to all datastore models rather than the more specific meaning that I think you want to convey.

I think we really need a third term here, but I can't think of one. "Activity" would be pretty good, except that it's already being used by subscriptions to mean "explorations or collections".

That said ... what if we say that an activity is an "exploration, collection, topic, story, question or skill" and then just say that subscriptions are only possible for the first two? That might be better conceptually.

If we go with that, could you add a clear comprehensive definition of "activity" somewhere, which provides a full list of what is considered an activity? Also add a note to the subscriptions model that only collections and explorations support subscriptions at the moment, and then maybe in future we can add that functionality for the other items. I think that should address the issue ... what do you think?

Reply:
Any idea when to mention the activity distinction? In the wiki? Cause I think it is used across the code and it is hard to find one place where to put it.

Reply:
Probably in any relevant storage model field descriptions. Would that work?

Reply:
Hmmm, we have one place where we use `activity_ids` I can mention it there, the issue is that we use `activity_mappings` even for stuff like suggestions or feedback, so it doesn't make sense to mention it there. I can maybe add it to the docstrings in the wipeout service too.

Reply:
Oh, that's a problem, because the term "activity" doesn't really apply to suggestions or feedback.

Could you please give a full list of the models in question? We might end up needing to call them pseudonymizable_entity_mappings or similar if that's what actually distinguishes them.

Reply:
Models in `config`, `collection`, `exploration`, `feedback`, `question`, `skill`, `story`, `suggestion`, `topic`.

Reply:
Wait, sorry, that's almost all the models. That's too broad. I meant the model classes, so that we can try to find a term for them that differentiates them from the rest.

Reply:
Done, changed to entity in all relevant places.